packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -264,8 +342,13 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
           pasteTimeoutRef.current = null;
         }, 500);
 
-        // Ensure we never accidentally interpret paste as regular input.
-        buffer.handleInput(key);
+        const content = key.sequence || '';
```

Comment:
move all of this logic into text buffer. that will work better to integrate it into the reducer.